Tensorify compute on Python scalars

bobrenjc93 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136147

Signed-off-by: Bob Ren <bobren@fb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping @ezyang w/ shipping dynamic float arguments in PT2

bobrenjc93 committed:
Tensorify compute on Python scalars

Signed-off-by: Bob Ren <bobren@fb.com>

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

ezyang commented on torch/fx/passes/tensorify_python_scalars.py:
```diff
@@ -0,0 +1,217 @@
+            nodes[i + 1] if node not in placeholders else first_non_placeholder
+        ):
+            # Look for tensor.item() calls on placeholders
+            if unbacked_bindings := node.meta.get("unacked_bindings"):
```
nit: unbacked_bindings

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Tensorify compute on Python scalars"


Signed-off-by: Bob Ren <bobrenfb.com>

Comandeered from https://github.com/pytorch/pytorch/pull/130228 as I'm helping ezyang w/ shipping dynamic float arguments in PT2


[ghstack-poisoned]
